src/language/dafnyInstallation.ts

```diff
@@ -89,15 +89,63 @@ export async function getCompilerRuntimePath(context: ExtensionContext): Promise
   return configuredPath;
 }
 
+// We cache the language server runtime path so that we don't need to copy it every time.
+let LanguageServerRuntimePath: string | null = null;
+
+function isNullOrEmpty(s: string | null): boolean {
+  return s === '' || s == null;
+}
+
 export async function getLanguageServerRuntimePath(context: ExtensionContext): Promise<string> {
-  const configuredPath = await ifNullOrEmpty(
-    getConfiguredLanguageServerRuntimePath(),
+  if(LanguageServerRuntimePath != null) {
+    return LanguageServerRuntimePath;
+  }
+  const configuredLanguageServerPath = getConfiguredLanguageServerRuntimePath();
+  const mightBeRecomputed: boolean = !isNullOrEmpty(configuredLanguageServerPath);
+
+  let configuredPath = await ifNullOrEmptyAsync(
+    configuredLanguageServerPath,
     async () => LanguageServerConstants.GetDefaultPath(await getConfiguredVersion(context))
   );
-  if(path.isAbsolute(configuredPath)) {
-    return configuredPath;
+  if(!path.isAbsolute(configuredPath)) {
+    configuredPath = path.join(context.extensionPath, configuredPath);
+  }
+  if(mightBeRecomputed) {
+    configuredPath = await cloneAllNecessaryDlls(configuredLanguageServerPath);
   }
-  return path.join(context.extensionPath, configuredPath);
+  LanguageServerRuntimePath = configuredPath;
+  return configuredPath;
+}
+
+async function cloneAllNecessaryDlls(configuredPath: string): Promise<string> {
+  const DLSName = 'DafnyLanguageServer';
+  const DCName = 'DafnyCore';
+  const dll = '.dll';
+  const DLS = DLSName + dll;
```

Comment:
nit: Keep the style-convention for local variables and start with lower-case letters. For example:
```suggestion
  const dlsName = 'DafnyLanguageServer';
  const dcName = 'DafnyCore';
  const dll = '.dll';
  const dls = dlsName + dll;
```